src/setting.rs

```diff
@@ -201,1380 +169,781 @@ pub enum SettingData {
-        let (input, lcd_brightness) = LCDBrightness::parse(input)?;
-        let (input, quantize) = Quantize::parse(input)?;
-        let (input, auto_cue_level) = AutoCueLevel::parse(input)?;
-        let (input, language) = Language::parse(input)?;
-        let (input, unknown2) = nom::number::complete::u8(input)?;
-        let (input, jog_ring_brightness) = JogRingBrightness::parse(input)?;
-        let (input, jog_ring_indicator) = JogRingIndicator::parse(input)?;
-        let (input, slip_flashing) = SlipFlashing::parse(input)?;
-        let (input, unknown3) = nom::bytes::complete::take(3usize)(input)?;
-        let unknown3 = unknown3.to_vec();
-        let (input, disc_slot_illumination) = DiscSlotIllumination::parse(input)?;
-        let (input, eject_lock) = EjectLock::parse(input)?;
-        let (input, sync) = Sync::parse(input)?;
-        let (input, play_mode) = PlayMode::parse(input)?;
-        let (input, quantize_beat_value) = QuantizeBeatValue::parse(input)?;
-        let (input, hotcue_autoload) = HotCueAutoLoad::parse(input)?;
-        let (input, hotcue_color) = HotCueColor::parse(input)?;
-        let (input, unknown4) = nom::number::complete::be_u16(input)?;
-        let (input, needle_lock) = NeedleLock::parse(input)?;
-        let (input, unknown5) = nom::number::complete::be_u16(input)?;
-        let (input, time_mode) = TimeMode::parse(input)?;
-        let (input, jog_mode) = JogMode::parse(input)?;
-        let (input, auto_cue) = AutoCue::parse(input)?;
-        let (input, master_tempo) = MasterTempo::parse(input)?;
-        let (input, tempo_range) = TempoRange::parse(input)?;
-        let (input, phase_meter) = PhaseMeter::parse(input)?;
-        let (input, unknown6) = nom::number::complete::be_u16(input)?;
-        let data = Self::MySetting {
-            unknown1,
-            on_air_display,
-            lcd_brightness,
-            quantize,
-            auto_cue_level,
-            language,
-            unknown2,
-            jog_ring_brightness,
-            jog_ring_indicator,
-            slip_flashing,
-            unknown3,
-            disc_slot_illumination,
-            eject_lock,
-            sync,
-            play_mode,
-            quantize_beat_value,
-            hotcue_autoload,
-            hotcue_color,
-            unknown4,
-            needle_lock,
-            unknown5,
-            time_mode,
-            jog_mode,
-            auto_cue,
-            master_tempo,
-            tempo_range,
-            phase_meter,
-            unknown6,
-        };
-        Ok((input, data))
-    }
 }
 
 /// Found at "PLAYER > DJ SETTING > PLAY MODE / AUTO PLAY MODE" of the "My Settings" page in the
 /// Rekordbox preferences.
 #[derive(Debug)]
+#[binread]
 pub enum PlayMode {
     /// Named "CONTINUE / ON" in the Rekordbox preferences.
+    #[br(magic = 0x80u8)]
     Continue,
     /// Named "SINGLE / OFF" in the Rekordbox preferences.
+    #[br(magic = 0x81u8)]
     Single,
```

Comment:
I think the preferred idiom is using a unit-like struct
```suggestion
#[derive(Debug)]
#[binread]
#[brw(repr(u8))]
pub enum PlayMode {
    /// Named "CONTINUE / ON" in the Rekordbox preferences.
    Continue = 0x80,
    /// Named "SINGLE / OFF" in the Rekordbox preferences.
    Single,
```

Reply:
https://github.com/Holzhaus/rekordcrate/pull/40/commits/d745bb71165cd4ee270332b269b94101a1e14b7c